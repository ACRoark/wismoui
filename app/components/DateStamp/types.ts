export type DateStampFormat = 'auto' | 'long' | 'short';
