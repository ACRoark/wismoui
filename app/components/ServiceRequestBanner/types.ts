export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
