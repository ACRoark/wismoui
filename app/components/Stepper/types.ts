import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
