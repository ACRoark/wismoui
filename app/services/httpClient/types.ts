export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
