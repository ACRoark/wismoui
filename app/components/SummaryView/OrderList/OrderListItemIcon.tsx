import React from 'react';

import StartIcon from 'images/StartIcon';
import StopIcon from 'images/StopIcon';
import OrderRequestType from 'types/OrderRequestType';

interface IOrderListItemIconProps {
  requestType: OrderRequestType;
}

const OrderListItemIcon: React.FC<IOrderListItemIconProps> = (props: IOrderListItemIconProps): React.ReactElement => {
  const { requestType } = props;

  switch (requestType) {
    case 'MIMO_START':
      return <StartIcon className="dte-wismo-order-list-item-icon" title="start order" />;
    case 'MIMO_STOP':
      return <StopIcon className="dte-wismo-order-list-item-icon" title="stop order" />;
    default:
      throw new Error('Invalid order type');
  }
};

export default OrderListItemIcon;
