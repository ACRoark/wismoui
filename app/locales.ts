export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
