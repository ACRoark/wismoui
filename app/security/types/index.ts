import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
