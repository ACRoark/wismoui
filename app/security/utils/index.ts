import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
