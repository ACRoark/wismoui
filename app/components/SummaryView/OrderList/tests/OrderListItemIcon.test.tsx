import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { OrderRequestType, OrderRequestTypes } from 'types';

import OrderListItemIcon from '../OrderListItemIcon';

const expectNoConsoleErrors = (language: string, requestType: OrderRequestType): void => {
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<OrderListItemIcon requestType={requestType} />, language);

  expect(spy).not.toHaveBeenCalled();
};

const expectComponentToMatchSnapshot = (language: string, requestType: OrderRequestType): void => {
  const tree = createSnapshotWithIntl(<OrderListItemIcon requestType={requestType} />, language);

  expect(tree).toMatchSnapshot();
};

describe('<OrderListItemIcon />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    OrderRequestTypes.forEach((requestType: OrderRequestType): void => {
      it(`should not log errors in console when rendered for '${requestType}' requests and the current language is ${language}`, (): void =>
        expectNoConsoleErrors(language, requestType));

      it(`should render correctly when the request type is '${requestType}' and the current language is ${language}`, (): void =>
        expectComponentToMatchSnapshot(language, requestType));
    });
  });
});
