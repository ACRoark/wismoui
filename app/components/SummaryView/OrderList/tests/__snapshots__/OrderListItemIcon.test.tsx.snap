// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderListItemIcon /> should render correctly when the request type is 'MIMO_START' and the current language is en 1`] = `
<svg
  className="dte-wismo-order-list-item-icon"
  fill="none"
  height="58"
  viewBox="0 0 34 58"
  width="34"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    start order
  </title>
  <path
    d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
    stroke="currentColor"
    strokeWidth="2"
  />
</svg>
`;

exports[`<OrderListItemIcon /> should render correctly when the request type is 'MIMO_START' and the current language is es 1`] = `
<svg
  className="dte-wismo-order-list-item-icon"
  fill="none"
  height="58"
  viewBox="0 0 34 58"
  width="34"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    start order
  </title>
  <path
    d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
    stroke="currentColor"
    strokeWidth="2"
  />
  <path
    d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
    stroke="currentColor"
    strokeWidth="2"
  />
</svg>
`;

exports[`<OrderListItemIcon /> should render correctly when the request type is 'MIMO_STOP' and the current language is en 1`] = `
<svg
  fill="none"
  height="59"
  viewBox="0 0 37 59"
  width="37"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    stop order
  </title>
  <g
    clipPath="url(#clip0)"
  >
    <path
      d="M0.782227 38.6582V16.3837C0.782227 14.0826 2.34671 12.2417 4.30231 12.2417H4.69343C6.64903 12.2417 8.21352 14.0826 8.21352 16.3837V29.638"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M8.13525 25.0358V8.37585C8.13525 6.07476 9.69974 4.23389 11.6553 4.23389H12.0465C14.0021 4.23389 15.5665 6.07476 15.5665 8.37585V25.1278"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M22.8413 26.9686V10.2167C22.8413 7.91558 24.4058 6.07471 26.3614 6.07471H26.7525C28.7081 6.07471 30.2726 7.91558 30.2726 10.2167V21.0778"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M36.2177 38.6585V25.2201C36.2177 22.919 34.6532 21.0781 32.6976 21.0781H32.3065C30.3509 21.0781 28.7864 22.919 28.7864 25.2201V27.7053C28.7864 30.5586 28.7864 31.387 26.909 32.4915C25.3446 33.412 23.5454 35.529 23.5454 37.738V39.3028"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M15.4883 24.6677V5.06238C15.4883 2.76128 17.0528 0.92041 19.0084 0.92041H19.3995C21.3551 0.92041 22.9196 2.76128 22.9196 5.06238V24.6677"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M36.2177 38.5664C36.2177 49.4276 28.3171 58.1717 18.5391 58.1717C8.76109 58.1717 0.782227 49.3355 0.782227 38.5664"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </g>
  <defs>
    <clipPath
      id="clip0"
    >
      <rect
        fill="white"
        height="59"
        width="37"
      />
    </clipPath>
  </defs>
</svg>
`;

exports[`<OrderListItemIcon /> should render correctly when the request type is 'MIMO_STOP' and the current language is es 1`] = `
<svg
  fill="none"
  height="59"
  viewBox="0 0 37 59"
  width="37"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    stop order
  </title>
  <g
    clipPath="url(#clip0)"
  >
    <path
      d="M0.782227 38.6582V16.3837C0.782227 14.0826 2.34671 12.2417 4.30231 12.2417H4.69343C6.64903 12.2417 8.21352 14.0826 8.21352 16.3837V29.638"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M8.13525 25.0358V8.37585C8.13525 6.07476 9.69974 4.23389 11.6553 4.23389H12.0465C14.0021 4.23389 15.5665 6.07476 15.5665 8.37585V25.1278"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M22.8413 26.9686V10.2167C22.8413 7.91558 24.4058 6.07471 26.3614 6.07471H26.7525C28.7081 6.07471 30.2726 7.91558 30.2726 10.2167V21.0778"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M36.2177 38.6585V25.2201C36.2177 22.919 34.6532 21.0781 32.6976 21.0781H32.3065C30.3509 21.0781 28.7864 22.919 28.7864 25.2201V27.7053C28.7864 30.5586 28.7864 31.387 26.909 32.4915C25.3446 33.412 23.5454 35.529 23.5454 37.738V39.3028"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M15.4883 24.6677V5.06238C15.4883 2.76128 17.0528 0.92041 19.0084 0.92041H19.3995C21.3551 0.92041 22.9196 2.76128 22.9196 5.06238V24.6677"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
    <path
      d="M36.2177 38.5664C36.2177 49.4276 28.3171 58.1717 18.5391 58.1717C8.76109 58.1717 0.782227 49.3355 0.782227 38.5664"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </g>
  <defs>
    <clipPath
      id="clip0"
    >
      <rect
        fill="white"
        height="59"
        width="37"
      />
    </clipPath>
  </defs>
</svg>
`;
